library/src/commonMain/kotlin/com/lagradost/cloudstream3/extractors/InternetArchive.kt

```diff
@@ -0,0 +1,112 @@
+package com.lagradost.cloudstream3.extractors
+
+import com.lagradost.cloudstream3.SubtitleFile
+import com.lagradost.cloudstream3.mvvm.logError
+import com.lagradost.cloudstream3.utils.ExtractorApi
+import com.lagradost.cloudstream3.utils.ExtractorLink
+import com.lagradost.cloudstream3.utils.Qualities
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import java.net.URLDecoder
+
+open class InternetArchive : ExtractorApi() {
+    override val mainUrl = "https://archive.org"
+    override val requiresReferer = false
+    override val name = "Internet Archive"
+
+    companion object {
+        private var archivedItems: MutableMap<String, Document> = mutableMapOf()
+
+        fun String.decodeUri(): String = URLDecoder.decode(this, "UTF-8")
+    }
+
+    override fun getExtractorUrl(id: String): String {
+        return "$mainUrl/details/$id"
+    }
+
+    override suspend fun getUrl(
+        url: String,
+        referer: String?,
+        subtitleCallback: (SubtitleFile) -> Unit,
+        callback: (ExtractorLink) -> Unit
+    ) {
+        val document = archivedItems[url] ?: run {
+            try {
+                val doc = Jsoup.connect(url).get()
```

Comment:
Is there a reason to use `Jsoup.connect(url).get()` as opposed to `app.get(url).document` ? 
Using the built in network client benefits from user settings (DNS) and caching.

Reply:
I can't remember why now lol. I created the extractor about 6 months ago before pushing here from where it was originally in my extension, and now I forgot why. I can try the other way.

Reply:
Done